### PR TITLE
ak-peter/traffic-light

### PR DESCRIPTION
**Pull request by Akshith-Patil**
## Proposed changes
New traffic light status
1. Replaced 'row' color status with a circular traffic light for rows and collapsible components
2. Minor CSS changes to black borders

## Types of changes

What types of changes does your code introduce?
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Further comments
**BEFORE**
Before - Status for Table
Before - Status for Collapsible

**AFTER** 
After - Light for Table
After - Light for collapsible

### CODE REVIEW

File: components/CollapsibleTable.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  );
+};
+
+function makeSubRow(
+  data: DependencyListSingleDep,
+  rank: number,
+  dependencyMap: DependencyMap
+) {
+  const dependencyData = dependencyMap.get(data.id);
+  const str =
```
Comment (alexdelicato) [inline components/CollapsibleTable.tsx:67]: could have a more informative var name

File: components/CollapsibleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            </TableRow>
+          </TableHead>
+          <TableBody>{rows}</TableBody>
+        </Table>
+      </TableContainer>
+    </div>
+  );
+};
+
+function makeSubRow(
```
Comment (alexdelicato) [inline components/CollapsibleTable.tsx:61]: small function description is probably a good idea

File: components/CollapsibleTable.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  let rowList: JSX.Element[] = [];
+
+  for (const dep of JSObject.deps) {
+    const data = JSObject.depMap.get(dep.id);
+    let internalSubRows: JSX.Element[] = [];
+    let externalSubRows: JSX.Element[] = [];
+    let userSubRows: JSX.Element[] = [];
+
+    let minRank = 2;
+
```
Comment (alexdelicato) [inline components/CollapsibleTable.tsx:119]: a short description of what this loop is / what ranks are might improve maintainability

File: components/Colours.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// import colours from '../global.css';
```
Comment (alexdelicato) [inline components/Colours.ts:1]: maybe delete this line if its not being used?

File: components/Row.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import { makeStyles } from "@mui/styles";
+
+import { Colours } from "./Colours";
+
+import Tabs from "./Tabs";
+import { DependencyData, semVerToString } from "./dataProcessing";
+
+import styles from "../components/row.module.css";
+import { Circle } from "@mui/icons-material";
+import { Typography } from "@mui/material";
```
Comment (alexdelicato) [inline components/Row.tsx:20]: might be beneficial to re-organise imports based on where they are coming from, i.e. mui, our components, etc.

File: components/Row.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          >
+            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+          </IconButton>
+        </TableCell>
+        {/* //The circle
+				<TableCell>
+						<svg viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
+							<circle cx="20" cy="20" r="20"/>
+						</svg>
+				</TableCell> */}
```
Comment (alexdelicato) [inline components/Row.tsx:78]: delete commented out code if not being used

File: components/Tabs.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+			{value === index && (
+				<Box sx={{ p: 3 }}>
+					<Typography>{children}</Typography>
+				</Box>
+			)}
+		</div>
+	);
+};
+
+const Tabs = (props) => {
```
Comment (alexdelicato) [inline components/Tabs.tsx:71]: some comments should probably be added to Tabs

File: components/dataProcessing.tsx [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+		skipMinor: skipMinor,
+		skipBug: skipBug
+	}
+}
+
+function externalLinkJSX(data: any): JSX.Element {
+	return <span style={{ float: "right" }}>(<a target="_blank" href={data.link}>GitHub</a>)</span>
+}
+
+export function findRank(used: SemVer, current: SemVer): number {
```
Comment (alexdelicato) [inline components/dataProcessing.tsx:71]: comments - what is the rank? why are we using it?

File: components/fonts.module.css [hunk 9]
```diff
@@ -0,0 +1,48 @@
+@font-face {
+	font-display: fallback;
+	font-family: 'proxima-nova';
+	font-style: normal;
+	font-weight: 100;
+	src: url('proximanova-thin.woff2') format('woff2'), url('proximanova-thin.woff') format('woff');
+  }
+  
+  @font-face {
```
Comment (alexdelicato) [inline components/fonts.module.css:9]: I think the alignment of the font-face is different to the top one

File: pages/_app.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import '../global.css'
+//TODO: Download/link to fonts
+//import '../components/fonts.module.css'
```
Comment (alexdelicato) [inline pages/_app.js:3]: remove line if not being used

**Review by alexdelicato — APPROVED** [review]
Looks good! You guys have done heaps of work well done. Most of my comments are about documentation and removing commented out code :)

**Comment by mt-empty** [comment]
@alexdelicato , are you approving or are you requesting comments?

**Comment by alexdelicato** [comment]
@mt-empty I approved it, the comments are just suggestions/feedback :)

File: components/dataProcessing.tsx (hunk 8 above)
Comment (Akshith-Patil) [reply components/dataProcessing.tsx:71]: rank is used for the traffic light. For example, if rank is 0 then the traffic light status is red, rank is 1 status is orange etc

File: components/Row.tsx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+    fontFamily: "Noto Sans, sans-serif",
+    fontSize: "large",
+  },
+});
+
+// Creates each invidial row
+export default function Row(props: {
+  rank: number;
+  row: any;
+  subRows: { internal: any[]; external: any[]; user: any[]; final: boolean };
```
Comment (mt-empty) [inline components/Row.tsx:33]: Are you sure you want to have `any` types here?

File: components/dataProcessing.tsx [hunk 12]
```diff
@@ -0,0 +1,373 @@
...
+		}
+	}
+	if (externalDeps.length > 0) {
+		sortedDeps.push(<li key="External">External:</li>);
+		for (const [rank, depList] of externalDeps) {
+			sortedDeps.push(depList);
+		}
+	}
+
+	//As we sorted based on rank, the minimum rank is at the first index
```
Comment (mt-empty) [inline components/dataProcessing.tsx:53]: what is `2`, and why its not its own constant

**Review by mt-empty — CHANGES_REQUESTED** [review]
Good job, I ran it and it look awesome. Here is my review:

1. Some of the filenames are capitalised why other are not, are you sure you want to do that?
2. Code is not formatted, you need to run pre-commit hooks. refer to https://github.com/ahm-monash/evergreen#readme=
3. It needs to pass the CI before merging, the documentation has a couple of types 
4. The branch name doesn't make sense

**Review by mt-empty — APPROVED** [review]

**Pull request merged**